src/services/itemMembership/plugins/MembershipRequest/service.ts

```diff
@@ -1,11 +1,60 @@
 import { singleton } from 'tsyringe';
 
+import { Member, PermissionLevel } from '@graasp/sdk';
+
+import { BaseLogger } from '../../../../logger';
+import { MAIL } from '../../../../plugins/mailer/langs/constants';
 import { MailerService } from '../../../../plugins/mailer/service';
+import { BUILDER_HOST } from '../../../../utils/config';
+import { Repositories } from '../../../../utils/repositories';
+import { Item } from '../../../item/entities/Item';
 
 @singleton()
 export class MembershipRequestService {
   private readonly mailerService: MailerService;
-  constructor(mailerService: MailerService) {
+  private readonly log: BaseLogger;
+
+  constructor(mailerService: MailerService, log: BaseLogger) {
     this.mailerService = mailerService;
+    this.log = log;
+  }
+
+  async get({ membershipRequestRepository }: Repositories, memberId: string, itemId: string) {
+    return await membershipRequestRepository.get(memberId, itemId);
+  }
+
+  async post({ membershipRequestRepository }: Repositories, memberId: string, itemId: string) {
+    return await membershipRequestRepository.post(memberId, itemId);
+  }
+
+  async notifyAdmins(member: Member, { itemMembershipRepository }: Repositories, item: Item) {
+    const adminMemberships = await itemMembershipRepository.getAllWithPermission(
+      item.path,
+      PermissionLevel.Admin,
+    );
+
+    const link = new URL(`/items/${item.id}/share`, BUILDER_HOST.url).toString();
```

Comment:
What do you think about building the url path using a function ? in this way, if the share url change, we just have to update the method ?